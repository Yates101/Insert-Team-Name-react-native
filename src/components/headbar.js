import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { LogoutButton } from './logoutButton';
import { useNavigation } from '@react-navigation/native';

function Headbar({loggedInStatus}) {

  const navigation = useNavigation();
  let logoutButton = null;

  // console.log(navigation.getParam('loggedInStatus'))
  console.log(loggedInStatus);

  if (loggedInStatus === true) {
    logoutButton = <LogoutButton />
  }

  // let logoutButton = null;
  // let loggedIn = false
  
  // fetch('http://localhost:3001/login', {
  //   method: 'POST',
  //   mode: 'cors',
  //   credentials: 'include',
  //   headers: {
  //     'Content-Type': 'application/json',
  //   },
  //   referrerPolicy: 'no-referrer',
  //   body: JSON.stringify( { user: {"username": userName, "password": passWord } } )
  //   })
  // .then((response) => response.json())
  // .then((json) => {
  //   if (json.logged_in === true) {
  //     loggedIn = true;
  //   } else {
  //     loggedIn = false;
  //   }
  //   })
  // .catch((error) => console.error(error));

  // if (loggedIn) {
  //     logoutButton = <LogoutButton />
  //   }

  return (
    <View style={styles.header}>
      <Text style={styles.text}>IkeBook</Text>
      {logoutButton}
    </View>
  )
}
      
const styles = StyleSheet.create({
  header: {
    alignSelf: 'stretch',
    height: 52,
    flexDirection: 'row', // row
    backgroundColor: '#179aff',
    alignItems: 'flex-end',
    justifyContent: 'space-between', // center, space-around
    paddingLeft: 10,
    paddingRight: 10
  },

  text: {
    fontSize: 32,
    fontWeight: 'bold',
    color: '#FFED00'
  }
})

export {Headbar};
