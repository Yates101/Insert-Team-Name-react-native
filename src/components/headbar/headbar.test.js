import 'react-native';
import React from 'react';

import { Headbar } from './headbar.js';
import renderer from 'react-test-renderer';

test('it works', () => {
  expect(true).toBeTruthy();
});

describe("<Headbar />", () => {
  const tree = renderer.create(<Headbar />).toJSON();

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  })

  it('has 1 children', () => {
    expect(tree.children.length).toBe(1);
  });
});
