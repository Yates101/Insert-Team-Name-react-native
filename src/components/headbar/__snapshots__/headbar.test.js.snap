// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headbar /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "alignSelf": "stretch",
      "backgroundColor": "#179aff",
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "space-between",
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFED00",
        "fontSize": 32,
        "fontWeight": "bold",
      }
    }
  >
    IkeBook
  </Text>
</View>
`;
